space/ring_circle.go

```diff
@@ -0,0 +1,110 @@
+package space
+
+import (
+	"math"
+
+	"github.com/spatial-go/geoos/algorithm/calc"
+)
+
+// Circle describes a circle Valid
+type Circle struct {
+	Ring
+	Centre   Point
+	Radius   float64
+	Segments int
+}
+
+// CreateCircle Returns valid circle.
+func CreateCircle(centre Point, radius float64) (*Circle, error) {
+	return CreateCircleWithSegments(centre, radius, calc.QuadrantSegments)
+}
+
+// CreateCircleWithSegments Returns valid circle.
+func CreateCircleWithSegments(centre Point, radius float64, segments int) (*Circle, error) {
+	circle := &Circle{Centre: centre, Radius: radius, Segments: segments}
+	circle.Ring = centre.Buffer(radius, segments).(Ring)
+	return circle, nil
+}
+
+// GeoJSONType returns the GeoJSON type for the circle.
+func (c *Circle) GeoJSONType() string {
+	return TypeCircle
```

Comment:
`Circle` is not a geometry type supported by GeoJSON. I think it should be `Polygon`.